From 851629f0065580eff884d6d3be435686d1d4b7e7 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 18 Mar 2020 13:26:37 +0200
Subject: [PATCH] [SW-10047] goya: pll: remove all config data

The PLLs configuration handling is moving to U-Boot in order to minimize
the preboot image.
Remove the PLLs handling from preboot.

Note: this slows down the Goya boot by about 4 seconds!

Change-Id: I8df58f1f56be33a268fc386900f1929955ff945e
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/goya/asic.c b/goya/asic.c
index 6501e8a..43b2513 100644
--- a/goya/asic.c
+++ b/goya/asic.c
@@ -13,69 +13,6 @@
 
 #include "common.h"
 
-static struct hl_pll_cfg goya_pll_cfg[] = {
-	{
-		.name = "IC", .base = mmIC_PLL_BASE,
-		.nr = 0, .nf = 63, .od = 15, .nb = 31,
-		.div_factor = { 0, 1, 1, 2 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
-		},
-	}, {
-		.name = "EMMC", .base = mmPSOC_EMMC_PLL_BASE,
-		.nr = 24, .nf = 415, .od = 15, .nb = 207,
-		.div_factor = { 0, 1, -1, -1 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK
-		},
-	}, {
-		.name = "MME", .base = mmPSOC_MME_PLL_BASE,
-		.nr = 0, .nf = 63, .od = 15, .nb = 31,
-		.div_factor = { 0, 1, 2, 8 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
-		},
-	}, {
-		.name = "PCI", .base = mmPSOC_PCI_PLL_BASE,
-		.nr = 4, .nf = 343, .od = 3, .nb = 171,
-		.div_factor = { 0, 35, 8, 85 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
-		},
-	}, {
-		.name = "TPC", .base = mmTPC_PLL_BASE,
-		.nr = 0, .nf = 63, .od = 15, .nb = 31,
-		.div_factor = { 0, 1, 1, 8 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
-		},
-	},
-};
-
-bool hl_pll_is_configurable(struct hl_pll_cfg *cfg)
-{
-	return cfg->base != mmCPU_PLL_BASE;
-}
-
-int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
-{
-	int pll_num = ARRAY_SIZE(goya_pll_cfg);
-
-	*hl_pll_cfg = goya_pll_cfg;
-
-	return pll_num;
-}
-
 u32 hl_skip_pcie_init(void)
 {
 	/* Use DumpSel[8] to skip PCIe initialization */
